source/portable/NetworkInterface/STM32/NetworkInterface.c

```diff
@@ -49,15 +49,15 @@
 #include "phyHandling.h"
 
 /* ST includes. */
-#if defined(STM32F4)
+#if defined(STM32F4xx)
     #include "stm32f4xx_hal_eth.h"
-#elif defined(STM32F7)
+#elif defined(STM32F7xx)
     #include "stm32f7xx_hal_eth.h"
-#elif defined(STM32H7)
+#elif defined(STM32H7xx)
     #include "stm32h7xx_hal_eth.h"
-#elif defined(STM32H5)
-    /* Untested */
-    #include "stm32h5xx_hal_eth.h"
+#elif defined(STM32H5xx)
+    /* #include "stm32h5xx_hal_eth.h" */
+    #include "stm32h5xx_hal.h"
```

Comment:
Including only the ETH module resulted in a lot of `unknown type name` errors. I think HAL libraries are designed so that you should always include the toplevel `stm32xxxx_hal.h` header

Reply:
Probably I did that because I was testing with all of the HAL drivers deleted except the eth driver, and used LL drivers instead.

Reply:
This is still an issue btw